### ISSUE TITLE
printing pauses after a few layers dependend on speed

### ISSUE THREAD

**Issue by hg42** [daid/LegacyCura#667]
I copied this from another issue, because it's a different problem:

I am running Cura on Linux (at 115200 baud).

Yesterday I pulled the current Cura code from github.

I wanted to print a part with about 2300 lines per layer.
The print seemed to stop at about the middle of the second layer.

It didn't really stop, but paused some seconds (burning a hole into the part) and then printed a single line of filament, paused again and so on. I finally canceled the print.

RepetierHost can print the part from saved gcode (but I didn't finish the print due to a mechanical problem, but the print went to about 10 layers without a problem).

I already had a somehow similar problem earlier in September:
issue #551 "Print window is crashing in the middle of all prints",
which I worked around by disabling all log prints and which was finally fixed by daid by limiting the number of lines in the log window.

In current Cura printWindow.py was replaced by printWindow2.py, which has borrowed the fixed part of code (in function _addTermLog) from printWindow.py.

So, currently I have no clue what could cause the problem.

I made some tests:
In September I managed to simulate a print by removing the power and by disabling both temperatures. So I could increase the print speed from 150 to 5000 which reduces testing time to seconds instead of minutes or hours.
Now, I repeated the simulation with my current problem.
The number of lines printed until the slow down depends on the printing speed, but it's not a linear dependency.
Faster speed gives fewer lines.

By the way, Cura always gave me a few errors in the log window, checksums etc. so there may be an additional problem here.
But before the update I could print large parts without noticable flaws.

Mechanically and electrically my printer proved to work and temperature values and other metrics are ok.
And because the gcode works, I think the problem is in the printing window or the communication with it.

**Comment by hg42** [comment]
seems I didn't understand the printWindow(2).py code...
These modules are imported at three places in the code. Two import peintWindow first and then printWindow2. The third imports only printWindow.

Both define some identical names e.g. the class printWindow. But printWindow2 lacks at least printProcessMonitor and startPrintInterface.
I'm not a "native" python programmer, so I can only guess, but I assume there is some redundancy here and this kind of mix may not be intended.
It seems printWindow2 should completely replace printWindow but it doesn't replace anything (I think) because "printWindow" is used as module name everywhere, e.g. in
  printWindow.startPrintInterface(options.printfile)
or
  connectionEntry.window = printWindow.printWindow(connection)
And indeed if I use different titles "Printing1" and "Printing2" in the modules I get a window titled "Printing1".

I tried to change all this by using
  from Cura.gui import printWindow2 as printWindow
which makes module printWindow2 available under the name printWindow (I think and this aliasing seems to work).
I also disable all
  from Cura.gui import printWindow
lines.
And of course I had to copy printProcessMonitor and startPrintInterface over to printWindow2.

Now I don't get any error, but I cannot start the print window (nothing happens when clicking the button), so I am still missing something?

**Comment by hg42** [comment]
It seems that printProcessMonitor shouldn't be used with printWindow2 and that's probably why it's not defined there. But disabling all occurances of _usbPrintMonitor doesn't help.
I think I'll give up for now (Christmas) and simply use printWindow without importing printWindow2 until this is resolved.

Have a nice Christmas everyone...

**Comment by daid** [comment]
printWindow2 is currently only used for the Doodle3D connection, I'm re-designing how the code interfaces with the printer, which is the printWindow2 implementation. USB printing is still on the normal printWindow.

**Comment by hg42** [comment]
Ok, thanks, and I'm very sorry, because it seems that I have an electrical problem just like the other issue.
I just noticed that the plug for the print head connection (hotend) is bend a little bit, when printing big parts.
My last print ran for 5 layers and then suddenly stopped with a min-temperature error.

But -- [thinking hard] -- my former observations don't fit in this scenario...there was no error message of this kind, and the print didn't stop, so may be I have two issues here.

I am currently investigating this further with another print, now with a fixed plug.

Btw. on my last print I disabled a "print" in printWindow.py (the one with ">" in front of the text), because this was also disabled in my previous working version of the file. So may be, this is allowed the print to go further than the first two layers. Lets see how the current print works...
